src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -454,6 +430,39 @@ protected function instantiateObject(array &$data, $class, array &$context, \Ref
         return new $class();
     }
 
+    /**
+     * @internal
+     */
+    protected function parseConstructorParameter(array &$data, $key, array &$context, \ReflectionParameter $constructorParameter, $format)
+    {
+        $parameterData = $data[$key];
+        if (null === $parameterData && $constructorParameter->allowsNull()) {
+            $params[] = null;
+            // Don't run set for a parameter passed to the constructor
+            unset($data[$key]);
+
+            return $parameterData;
+        }
+        try {
+            if (null !== $constructorParameter->getClass()) {
+                if (!$this->serializer instanceof DenormalizerInterface) {
+                    throw new LogicException(sprintf('Cannot create an instance of %s from serialized data because the serializer inject in "%s" is not a denormalizer', $constructorParameter->getClass(), self::class));
```

Comment:
`'Cannot create an instance of "%s" from serialized data because the serializer injected in "%s" is not a denormalizer'.`